File: VB/RegistrationForm.Lesson4/ViewModel/RegistrationViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.POCO
Imports RegistrationForm.DataModel
Imports System

Namespace RegistrationForm.ViewModel
    <POCOViewModel> _
    Public Class RegistrationViewModel
        Public Shared Function Create() As RegistrationViewModel
            Return ViewModelSource.Create(Function() New RegistrationViewModel())
        End Function
        Protected Sub New()
            MinBirthday = New Date(Date.Now.Year - 100, 12, 31)
            MaxBirthday = New Date(Date.Now.Year - 1, 12, 31)
            If Me.IsInDesignMode() Then
                InitializeInDesignMode()
            Else
                InitializeInRuntime()
            End If
        End Sub
        Private Sub InitializeInDesignMode()
            FirstName = "John"
            LastName = "Smith"
            Email = "John.Smith@JohnSmithMail.com"
            Password = "Password"
            ConfirmPassword = "Password"
            Birthday = New Date(1980, 1, 1)
            Gender = 1
        End Sub
        Private Sub InitializeInRuntime()
            Birthday = Nothing
            Gender = -1
        End Sub

        Public Overridable Property FirstName() As String
        Public Overridable Property LastName() As String
        Public Overridable Property Email() As String
        Public Overridable Property Password() As String
        Public Overridable Property ConfirmPassword() As String
        Public Overridable Property Birthday() As Date?
        Public Overridable Property MinBirthday() As Date
        Public Overridable Property MaxBirthday() As Date
        Public Overridable Property Gender() As Integer

        Public Sub AddEmployee()
            EmployeesModelHelper.AddNewEmployee(FirstName, LastName, Email, Password, Birthday.Value, Gender)
        End Sub
    End Class
End Namespace

File: CS/RegistrationForm.Lesson2/ViewModel/RegistrationViewModel.cs
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.POCO;
using RegistrationForm.DataModel;
using System;

namespace RegistrationForm.ViewModel {
    [POCOViewModel]
    public class RegistrationViewModel {
        public static RegistrationViewModel Create() {
            return ViewModelSource.Create(() => new RegistrationViewModel());
        }
        protected RegistrationViewModel() {
            if(this.IsInDesignMode())
                InitializeInDesignMode();
            else InitializeInRuntime();
        }
        void InitializeInDesignMode() {
            FirstName = "John";
            LastName = "Smith";
            Email = "John.Smith@JohnSmithMail.com";
            Password = "Password";
            ConfirmPassword = "Password";
            Birthday = new DateTime(1980, 1, 1);
            Gender = 1;
        }
        void InitializeInRuntime() {
            Birthday = null;
            Gender = -1;
        }

        public virtual string FirstName { get; set; }
        public virtual string LastName { get; set; }
        public virtual string Email { get; set; }
        public virtual string Password { get; set; }
        public virtual string ConfirmPassword { get; set; }
        public virtual DateTime? Birthday { get; set; }
        public virtual int Gender { get; set; }

        public void AddEmployee() {
            EmployeesModelHelper.AddNewEmployee(FirstName, LastName, Email, Password, Birthday.Value, Gender);
        }
    }
}